core/src/main/java/com/datastrato/gravitino/metrics/source/MetricsSource.java

```diff
@@ -85,6 +97,15 @@ public Histogram getHistogram(String name) {
    * @return a new or pre-existing Timer
    */
   public Timer getTimer(String name) {
-    return this.metricRegistry.timer(name);
+    return this.metricRegistry.timer(
+        name,
+        () ->
+            new Timer(
+                new SlidingTimeWindowArrayReservoir(
+                    getTimeSlidingWindowSeconds(), TimeUnit.SECONDS)));
+  }
+
+  protected int getTimeSlidingWindowSeconds() {
```

Comment:
Why do you need this, can you directly use the class variable?

Reply:
`HttpServerMetricsSource` should use this method, you suggest to make the variable protected and use it directly?

Reply:
it's fine leave as it is.